src/chainstate/stacks/address.rs

```diff
@@ -49,46 +49,246 @@ use crate::types::chainstate::STACKS_ADDRESS_ENCODED_SIZE;
 use crate::util_lib::boot::boot_code_addr;
 
 pub trait StacksAddressExtensions {
-    fn to_bitcoin_tx_out(&self, value: u64) -> TxOut;
-    fn as_clarity_tuple(&self) -> TupleData;
     fn to_b58(self) -> String;
     fn from_bitcoin_address(addr: &BitcoinAddress) -> StacksAddress;
     fn is_boot_code_addr(&self) -> bool;
 }
 
-impl StacksAddressExtensions for StacksAddress {
-    /// is this a boot code address, if the supplied address is mainnet or testnet,
-    ///  it checks against the appropriate the boot code addr
-    fn is_boot_code_addr(&self) -> bool {
-        self == &boot_code_addr(self.is_mainnet())
+/// A PoX address as seen by the .pox and .pox-2 contracts.
+/// Used by the sortition DB and chains coordinator to extract addresses from the PoX contract to
+/// build the reward set and to validate block-commits.
```

Comment:
Maybe add "This comprises a larger set of possible addresses than `StacksAddress`"?

Reply:
Sure, will do

Reply:
Done